### ISSUE TITLE
Objective Sharpie - fails if Cocoapods not installed globally

### ISSUE THREAD

**Issue by J-Swift** [xamarin/xamarin-macios#15916]
I assume sharpie is doing a `which pod` to find the global install, but I think it should also consider `bundle exec which pod` for the common case of directory-isolated gems.

### Steps to Reproduce

1. Install cocopods locally (i.e. `bundle install` with a BUNDLE_PATH specified)
2. Run `sharpie pod init Foo`

### Expected Behavior

It finds the bundled cocoapods and runs

### Actual Behavior

It fails with

```
error: CocoaPods is not installed

>> install it by running sudo gem install cocoapods
>> read more about CocoaPods at https://cocoapods.org
```

I'm able to fake it out by putting a shell script called `pod` in my PATH with these contents:

```
#!/usr/bin/env bash

bundle exec pod $@
```

### Environment

- sharpie v3.5.61-c2b0b612
- bundler v2.3.7
- ruby v3.1.2p20

**Comment by chamons** [comment]
Thanks for the feedback. 

I suspect that sharpie support for cocoa pods is deprecated, so your work around is likely the best path forward. 

Hey @dalexsoto - can you confirm sharpie pod support is deprecated?

**Comment by J-Swift** [comment]
Yeah the script "works" but the next commands end up hanging for, as far as I can tell, forever. I left the computer for several hours and came back but it never finished the `init` process.
